libs/forms/core/src/dyn-providers.ts

```diff
@@ -1,17 +1,30 @@
 import { Validators } from '@angular/forms';
+import { of } from 'rxjs';
+import { map, startWith } from 'rxjs/operators';
+import { DynConfigId } from './control-config.types';
+import { DynControlCondition, DynControlConditionFn, DynControlMatchCondition, DynControlMatcher, DynControlMatcherFn } from './control-matchers.types';
 import { DynValidatorProvider } from "./control-validation.types";
+import { DynTreeNode } from './tree.types';
 
 /**
- * Base provider
+ * Base types
  */
 export interface DynBaseProvider {
   priority?: number;
 }
 
+export type DynHandlerFactory<F> = (...args: any[]) => F;
+
+export interface DynBaseHandler<F> extends DynBaseProvider {
+  id: DynConfigId;
+  fn: DynHandlerFactory<F>;
+}
```

Comment:
the Handlers are provided via TOKENs with an `id` and `priority` (to override any previous one),
and then Maps them by `<id, FnFactory>` to be used according the configured `[id, args]`